core/controllers/topic_editor.py

```diff
@@ -183,8 +186,17 @@ def get(self, topic_id):
             raise self.PageNotFoundException(
                 Exception('The topic with the given id doesn\'t exist.'))
 
+        skill_ids = copy.deepcopy(topic.uncategorized_skill_ids)
```

Comment:
The logic of lines 189-192 is repeated. Make a method in the Topic object (topic.get_all_skill_ids()) and add comprehensive unit tests for it.

Reply:
Done.